exe/Daemon.hs

```diff
@@ -305,30 +305,35 @@ loadMachine mode id = do
   (idx, is) <- machineIpfs id $ machineInputsFrom id Nothing
   sub <- initDaemonSubscription id
   m <- emptyMachine id mode sub
-  (m', _) <- addInputs is idx m
+  m' <- addInputs is idx m
   insertMachine m'
   pure m'
 
--- | Add inputs to a machine and update its state
-addInputs
-  :: [Value]
-  -- ^ Inputs to add.
-  -> MachineEntryIndex
-  -- ^ The new index.
-  -> Machine
-  -> Daemon (Machine, [Value])
-  -- ^ Returns the updated machine and the produced outputs
-addInputs inputs index m = do
+-- | Checks that inputs can be applied to a machine. Returns the new state that
+-- would result, and the outputs.
+checkInputs :: Machine -> [Value] -> Daemon (Bindings (PrimFns Identity), [Value])
```

Comment:
How about calling this `runInputs`? And update the docs to mention that it throws when an error when evaluation fails.